### PR TITLE
Add user assign functionality to Role Edit

### PR DESCRIPTION
**Pull request by DimalChandrasiri**

### CODE REVIEW

File: apps/admin-portal/src/components/roles/edit-role/edit-role-users.tsx [hunk 1]
```diff
@@ -16,28 +16,92 @@
...
+import { useTranslation } from "react-i18next";
+import { useDispatch } from "react-redux";
+import { addAlert } from "../../../store/actions";
+
+interface RoleUserDetailsProps {
+    roleObject: RolesInterface;
+    onRoleUpdate: () => void;
+}
+
+export const RoleUserDetails: FunctionComponent<RoleUserDetailsProps> = (props: RoleUserDetailsProps): ReactElement => {
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/components/roles/edit-role/edit-role-users.tsx]: Can we split this into two lines?

File: apps/admin-portal/src/components/roles/create-role-wizard/role-permisson.tsx [hunk 2]
```diff
@@ -115,8 +116,8 @@ export const PermissionList: FunctionComponent<PermissionListProp> = (props: Per
     useEffect(() => {
         setPermissionsLoading(true);
 
-        if (role) {
-            getPermissionsPerRole(role).then(response => {
+        if (roleObject) {
+            getPermissionsPerRole(roleObject.id).then(response => {
```
Comment (brionmario) [inline apps/admin-portal/src/components/roles/create-role-wizard/role-permisson.tsx]: Shall we bring `.then` to a new line.

File: apps/admin-portal/src/components/roles/create-role-wizard/role-user-assign.tsx [hunk 3]
```diff
@@ -49,6 +53,18 @@ export const AddRoleUsers: FunctionComponent<AddRoleUserProps> = (props: AddRole
         getUsersList(limit, offset, filter, attribute, null)
             .then((response) => {
                 setUsersList(response.Resources);
+
+                if (assignedUsers && assignedUsers.length != 0) {
```
Comment (brionmario) [inline apps/admin-portal/src/components/roles/create-role-wizard/role-user-assign.tsx]: Better to use a strict check here. i.e `!==`

File: apps/admin-portal/src/components/roles/create-role-wizard/role-user-assign.tsx [hunk 4]
```diff
@@ -49,6 +53,18 @@ export const AddRoleUsers: FunctionComponent<AddRoleUserProps> = (props: AddRole
...
+                    const selectedUserList: UserBasicInterface[] = [];
+                    response.Resources.forEach(user => {
```
Comment (brionmario) [inline apps/admin-portal/src/components/roles/create-role-wizard/role-user-assign.tsx]: Would be better to check if `Resources` exists in the response and it's an iterable array.

File: apps/admin-portal/src/components/roles/edit-role/edit-role-basic.tsx [hunk 5]
```diff
@@ -85,11 +85,20 @@ export const BaiscRoleDetails: FunctionComponent<BasicRoleProps> = (props: Basic
...
+            Operations: [{
+                "op": "replace",
+                "value": {
+                    "displayName": values.get("rolename")
+                }
+            }]
         }
 
-        updateRoleDetails(roleId, roleData).then(response => {
+        updateRoleDetails(roleObject.id, roleData).then(response => {
```
Comment (brionmario) [inline apps/admin-portal/src/components/roles/edit-role/edit-role-basic.tsx]: Shall we bring `then` to a new line and handle `catch` as well.

File: apps/admin-portal/src/models/roles.ts [hunk 6]
```diff
@@ -88,3 +88,13 @@ export interface SearchRoleInterface {
     startIndex: number;
     filter: string;
 }
+
+export interface PatchRoleData {
```
Comment (brionmario) [inline apps/admin-portal/src/models/roles.ts]: Shall we add `Interface` prefix here.

Comment (DimalChandrasiri) [reply apps/admin-portal/src/models/roles.ts]: fixed with commit 0482c99

File: apps/admin-portal/src/components/roles/edit-role/edit-role-basic.tsx (hunk 5 above)
Comment (DimalChandrasiri) [reply apps/admin-portal/src/components/roles/edit-role/edit-role-basic.tsx]: fixed with commit 0482c99

File: apps/admin-portal/src/components/roles/create-role-wizard/role-user-assign.tsx (hunk 4 above)
Comment (DimalChandrasiri) [reply apps/admin-portal/src/components/roles/create-role-wizard/role-user-assign.tsx]: fixed with commit 0482c99

File: apps/admin-portal/src/components/roles/create-role-wizard/role-user-assign.tsx (hunk 3 above)
Comment (DimalChandrasiri) [reply apps/admin-portal/src/components/roles/create-role-wizard/role-user-assign.tsx]: fixed with commit 0482c99

File: apps/admin-portal/src/components/roles/create-role-wizard/role-permisson.tsx (hunk 2 above)
Comment (DimalChandrasiri) [reply apps/admin-portal/src/components/roles/create-role-wizard/role-permisson.tsx]: fixed with commit 0482c99

File: apps/admin-portal/src/components/roles/edit-role/edit-role-users.tsx (hunk 1 above)
Comment (DimalChandrasiri) [reply apps/admin-portal/src/components/roles/edit-role/edit-role-users.tsx]: fixed with commit 0482c99

File: apps/admin-portal/src/pages/role-edit.tsx [hunk 7]
```diff
@@ -39,6 +39,10 @@ export const RoleEditPage: FunctionComponent<any> = (props: any): ReactElement =
         })
     }
 
+    const onRoleUpdate = (): void => {
+        getRoleDetails(roleId);
+    }
```
Comment (jeradrutnam) [inline apps/admin-portal/src/pages/role-edit.tsx]:
```suggestion
    };
```

File: apps/admin-portal/src/components/roles/edit-role/edit-role-users.tsx [hunk 8]
```diff
@@ -16,28 +16,94 @@
...
+
+    /**
+     * Dispatches the alert object to the redux store.
+     *
+     * @param {AlertInterface} alert - Alert object.
+     */
+    const handleAlerts = (alert: AlertInterface) => {
+        dispatch(addAlert(alert));
+    };
+
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/roles/edit-role/edit-role-users.tsx]:
```suggestion
```

**Pull request merged**